contrib/epee/include/net/net_utils_base.h

```diff
@@ -107,6 +107,54 @@ namespace net_utils
 	inline bool operator>=(const ipv4_network_address& lhs, const ipv4_network_address& rhs) noexcept
 	{ return !lhs.less(rhs); }
 
+	class ipv6_network_address
+	{
+		std::string m_ip;
```

Comment:
Why std::string, rather than an array of however number of bytes ? The string means it's never certain to be canonical.

Reply:
changed to using boost's address_v6 under the hood

Reply:
Looks good, any chance the m_address_str that's still left can be made canonical ? Otherwise stuff that uses the host string as a key (ie, the block list) will still see the same address as different.

Reply:
It looks a bit weird, but yes (and it should be fine despite the appearance), and done.